src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -0,0 +1,215 @@
+    private final int minSize;
+    private final int maxSize;
+
+    MapMutator(SerializingMutator<K> keyMutator, SerializingMutator<V> valueMutator, int minSize,
+        int maxSize) {
+      this.keyMutator = keyMutator;
+      this.valueMutator = valueMutator;
+      this.minSize = minSize;
+      this.maxSize = maxSize;
+    }
+
+    @Override
+    public Map<K, V> read(DataInputStream in) throws IOException {
+      int size = Math.max(in.readInt(), 0);
+      Map<K, V> map = new HashMap<>(size);
+      for (int i = 0; i < size; i++) {
+        map.put(keyMutator.read(in), valueMutator.read(in));
+      }
+      // Wrap in an immutable view for additional protection against accidental mutation in fuzz
+      // tests.
+      return toImmutableMapView(map);
+    }
+
+    @Override
+    public void write(Map<K, V> map, DataOutputStream out) throws IOException {
+      out.writeInt(map.size());
+      for (Map.Entry<K, V> entry : map.entrySet()) {
+        keyMutator.write(entry.getKey(), out);
+        valueMutator.write(entry.getValue(), out);
+      }
+    }
+
+    @Override
+    protected Map<K, V> makeDefaultInstance() {
+      // Wrap in an immutable view for additional protection against accidental mutation in fuzz
+      // tests.
+      return toImmutableMapView(new HashMap<>(maxInitialSize()));
+    }
+
+    @Override
+    public void initInPlace(Map<K, V> reference, PseudoRandom prng) {
+      int targetSize = prng.closedRange(minInitialSize(), maxInitialSize());
+      Map<K, V> map = underlyingMutableList(reference);
+      map.clear();
+      for (int i = 0; i < targetSize; i++) {
+        map.put(keyMutator.init(prng), valueMutator.init(prng));
+      }
+    }
+
+    @Override
+    public void mutateInPlace(Map<K, V> reference, PseudoRandom prng) {
+      Map<K, V> map = underlyingMutableList(reference);
+      if (map.isEmpty()) {
+        map.put(keyMutator.init(prng), valueMutator.init(prng));
+      } else if (!prng.trueInOneOutOf(4)) {
+        // Choose a random entry, and then key or value, to mutate.
+        int i = prng.indexIn(map.size());
+        map.entrySet().stream().skip(i).findFirst().ifPresent(entry -> {
+          K key = entry.getKey();
+          V value = entry.getValue();
+          map.remove(key);
+          if (prng.choice()) {
+            key = keyMutator.mutate(key, prng);
+          } else {
+            value = valueMutator.mutate(value, prng);
+          }
+          map.put(key, value);
+        });
+      } else {
+        // This will increase the map size up to max and then oscillate around it.
+        // Smaller sizes are already used, so increasing the size should be a valid choice.
+        if (map.size() < maxSize) {
+          // Create a new entry, as a deep copy could be expensive.
+          map.put(keyMutator.init(prng), valueMutator.init(prng));
+        } else if (map.size() > minSize) {
+          // Maps are not ordered, removing the first one is as good as any.
```

Comment:
Not sure that's true. I don't know how Java randomizes the map order exactly, but it's possible that it stays the same within one JVM process. Maybe we should again traverse a random number of elements before removing one?

Reply:
Done.